sdk/playwrighttesting/Azure.Developer.MicrosoftPlaywrightTesting.TestLogger/src/Processor/TestProcessor.cs

```diff
@@ -335,13 +333,13 @@ private void GenerateMarkdownSummary()
 
 ![skipped](https://img.shields.io/badge/status-skipped-lightgrey) **Skipped:** {SkippedTestCount}
 
-#### For more details, visit the [service dashboard]({Uri.EscapeUriString(_cloudRunMetadata.PortalUrl!)}).
+#### For more details, visit the [service dashboard]({Uri.EscapeDataString(_cloudRunMetadata.PortalUrl!)}).
 ";
 
-                string filePath = Environment.GetEnvironmentVariable("GITHUB_STEP_SUMMARY");
+                string? filePath = Environment.GetEnvironmentVariable("GITHUB_STEP_SUMMARY");
                 try
                 {
-                    File.WriteAllText(filePath, markdownContent);
+                    File.WriteAllText(filePath ?? string.Empty, markdownContent);
```

Comment:
What happens if you try to write data to a file where the file path is specified as an empty string?